web/web/api/v1/channel.py

```diff
@@ -0,0 +1,37 @@
+from flask import request, jsonify, Blueprint
+from marshmallow import ValidationError
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound, MultipleResultsFound
+
+from .response_wrapper import ApiResponseWrapper
+from web.database import db
+from web.models.channel import Channel, ChannelSchema
+
+channel_api_bp = Blueprint('channel_api_bp', __name__)
+
+@channel_api_bp.route('/channel', methods=['POST'])
+def add_channel():
+    '''
+    Adds new channel object to database
+    '''
+
+    arw = ApiResponseWrapper()
+    channel_schema = ChannelSchema(exclude=['channel_id'])
+    new_channel = request.get_json()
+
+    try:
+        new_channel = channel_schema.load(new_channel, session=db.session)
+        db.session.add(new_channel)
+        db.session.commit()
+
+    except IntegrityError:
+        db.session.rollback()
+        arw.add_errors('Conflict while loading data')
+        return arw.to_json(None, 400)
+
+    except ValidationError as ve:
+        arw.add_errors(ve.messages)
```

Comment:
do you need to rollback on validation error as well? Or only integrity error?

Reply:
Yes, validation and integrity error should both be used, except I think it makes more sense for the validation check to happen before the integrity check. Validation error checks to make sure the information in the json object is accepted by the marshmallow schema, so if one of the keys isn't there, it'll return " [missing key name]: Missing key name" or if there is an incorrect name (say HomeHub, which doesn't have a meter id foreign key, but the json object in the post route has a meter_id in the body), then it'll provide the error:          "meter_id": [
                "Unknown field."
            ]

I checked again without the validation error, it errors out without the feedback. Integrity error, on the other hand, checks to make sure the values provided are correct according to the database when it is being added to the database.

Reply:
Yes I think it would be a good idea to have a rollback here since db.session.add is included in the try statement. I also think validation check should be before the integrity check since validation is making sure the information aligns with the marshmallow schema, and integrity makes sure that it ok to add to the database.